crates/exex/src/hints.rs

```diff
@@ -50,6 +63,21 @@ impl KakarotHintProcessor {
         self
     }
 
+    /// Adds a block to the [`KakarotHintProcessor`].
+    ///
+    /// This method allows you to register a block by providing a [`SealedBlockWithSenders`]
+    /// instance.
+    pub fn with_block_and_transaction(
+        mut self,
+        block: SealedBlock,
+        transaction: TransactionSignedEcRecovered,
+    ) -> Self {
+        self = self.with_hint(block_info_hint(block.clone(), transaction.clone()));
+        self.block = Some(block.clone());
```

Comment:
do we need to store the block here? I don't see it used elsewhere

Reply:
I just deleted them because they weren't used indeed. Actually, I think we can do better but I need to think about it in a follow up because it would be more practical to have the blocks as hint processor fields to use them in the hints rather than having the hints outside and passing arguments (that was my initial idea). But I need to think about it specifically to do something clean.